packages/explorer/src/store.ts

```diff
@@ -0,0 +1,59 @@
+import { Hex, fromHex } from "viem";
+import { create } from "zustand";
+import { ACCOUNTS } from "./consts";
+
+type Balances = {
+  [key: string]: bigint;
+};
+
+type BalanceResponse = {
+  id: number;
+  result: Hex;
+};
+
+type Store = {
+  account: Hex;
+  setAccount: (account: Hex) => void;
+
+  balances: Balances;
+  fetchBalances: () => void;
+};
+
+export const useStore = create<Store>((set) => ({
+  account: ACCOUNTS[0],
+  setAccount: (account) => set({ account }),
+
+  balances: {},
+  fetchBalances: async () => {
```

Comment:
ooc why this instead of a component for an account balance, using wagmi hook, then iterating through accounts -> balance in react tree?

Reply:
Wagmi can only fetch one balance at a time, no? Couldn't find how to fetch several balances all at once. The nice micro-optimization with this one is that all balances are fetched in one POST request.

Reply:
you can enable batch RPC calls in your viem/wagmi client

then do something like (pseudocode)

```tsx
<>
  <h1>Accounts</h1>
  {accounts.map((account) => (
    <div key={account}>{account}: <AccountBalance account={account} /></div>
  )}
</>
```

```tsx
function AccountBalance({ account }) {
  const balance = useBalance({ account });
  return <>{formatEther(balance)}</>;
```

Reply:
Was a bit hesitant at first but after refactoring I think it's much nicer now